### PR TITLE
Typo-squatting: finish functional tests

### PR DESCRIPTION
**Pull request by zhhyu**
Finish the basic typo-squatting functional tests. 

Fix: https://github.com/NuGet/Engineering/issues/1595

### CODE REVIEW

File: tests/NuGetGallery.FunctionalTests/TyposquattingCheck/TyposquattingCheckForUploadPackageTests.cs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        private readonly CommandlineHelper _commandlineHelper;
+        private readonly PackageCreationHelper _packageCreationHelper;
+        public TyposquattingCheckForUploadPackageTests(ITestOutputHelper testOutputHelper) : base(testOutputHelper)
+        {
+            _commandlineHelper = new CommandlineHelper(TestOutputHelper);
+            _packageCreationHelper = new PackageCreationHelper(testOutputHelper);
+        }
+
+        [TyposquattingTestFact]
+        public async Task UploadNotTyposquattingPackage()
```
Comment (joelverhagen) [inline tests/NuGetGallery.FunctionalTests/TyposquattingCheck/TyposquattingCheckForUploadPackageTests.cs]: Spoke offline, this test is pretty useless, given there are many other tests that verify push succeeds.

File: tests/NuGetGallery.FunctionalTests/TyposquattingCheck/TyposquattingCheckForUploadPackageTests.cs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        }
+
+        [TyposquattingTestFact]
+        public async Task UploadTyposquattingPackageAndBlockUser()
+        {
+            var packageId = "newtonsoft-json";
+            string version = "1.0.0";
+            string packageFullPath = await _packageCreationHelper.CreatePackageWithMinClientVersion(packageId, version, "2.3");
+
+            var processResult = await _commandlineHelper.UploadPackageAsync(packageFullPath, UrlHelper.V2FeedPushSourceUrl);
```
Comment (joelverhagen) [inline tests/NuGetGallery.FunctionalTests/TyposquattingCheck/TyposquattingCheckForUploadPackageTests.cs]: Let's assert the message is related to typo-squatting so we know we are testing the right code path.

File: tests/NuGetGallery.FunctionalTests/TyposquattingCheck/TyposquattingCheckForUploadPackageTests.cs (hunk 1 above)
Comment (zhhyu) [reply tests/NuGetGallery.FunctionalTests/TyposquattingCheck/TyposquattingCheckForUploadPackageTests.cs]: Delete it.

File: tests/NuGetGallery.FunctionalTests/TyposquattingCheck/TyposquattingCheckForUploadPackageTests.cs (hunk 2 above)
Comment (zhhyu) [reply tests/NuGetGallery.FunctionalTests/TyposquattingCheck/TyposquattingCheckForUploadPackageTests.cs]: Assert message now.

File: tests/NuGetGallery.FunctionalTests.Core/XunitExtensions/TyposquattingTestFactAttribute.cs [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+namespace NuGetGallery.FunctionalTests.XunitExtensions
+{
+    public class TyposquattingTestFactAttribute : FactAttribute
+    {
+        public TyposquattingTestFactAttribute()
+        {
+            if (!GalleryConfiguration.Instance.TyposquattingCheckAndBlockUsers)
+            {
+                Skip = string.Format("Typosquatting checking or user blocking are disabled");
```
Comment (agr) [inline tests/NuGetGallery.FunctionalTests.Core/XunitExtensions/TyposquattingTestFactAttribute.cs]: No need for `string.Format` here.

Comment (zhhyu) [reply tests/NuGetGallery.FunctionalTests.Core/XunitExtensions/TyposquattingTestFactAttribute.cs]: Delete it now.

**Pull request merged**